core/src/main/java/io/questdb/griffin/engine/functions/json/JsonPathVarcharFunc.java

```diff
@@ -0,0 +1,210 @@
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.json;
+
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.VarcharFunction;
+import io.questdb.std.str.DirectUtf8Sequence;
+import io.questdb.std.str.DirectUtf8Sink;
+import io.questdb.std.str.Utf8Sequence;
+import io.questdb.std.str.Utf8Sink;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+class JsonPathVarcharFunc extends VarcharFunction implements BinaryFunction, JsonPathFunc {
+    private final VarcharSupportingState a;
+    private final VarcharSupportingState b;
+    private final VarcharSupportingState copied;
+    private final Function json;
+    private final int maxSize;
+    private final Function path;
+    private final DirectUtf8Sink pointer;
+    private final boolean strict;
+    private DirectUtf8Sink defaultVarchar = null;
+
+    public JsonPathVarcharFunc(Function json, Function path, DirectUtf8Sink pointer, int maxSize, boolean strict) {
+        this.a = new VarcharSupportingState(new DirectUtf8Sink(maxSize));
+        this.b = new VarcharSupportingState(new DirectUtf8Sink(maxSize));
+        this.copied = new VarcharSupportingState();
+        this.json = json;
+        this.path = path;
+        this.pointer = pointer;
+        this.maxSize = maxSize;
+        this.strict = strict;
+    }
+
+    @Override
+    public void close() {
+        a.close();
+        b.close();
+        copied.close();
+        pointer.close();
+        if (defaultVarchar != null) {
+            defaultVarchar.close();
+        }
+    }
+
+    @Override
+    public Function getLeft() {
+        return json;
+    }
+
+    @Override
+    public String getName() {
+        return "json_path";
+    }
+
+    @Override
+    public Function getRight() {
+        return path;
+    }
+
+    @Override
+    public void getVarchar(Record rec, Utf8Sink utf8Sink) {
+        if (utf8Sink instanceof DirectUtf8Sink) {
+            if ((copied.destSink != null) && copied.closeDestSink) {
+                copied.destSink.close();
```

Comment:
No-op since IntelliJ correctly spots that the value is overwritten anyways the line below to a value. Misc.free is not helpful in this context.
Marking as resolved.